controller/pokemon.go

```diff
@@ -1,15 +1,74 @@
 package controller
 
 import (
-	"encoding/json"
-	"fmt"
-	"net/http"
 	"bootcamp/domain/model"
 	"bootcamp/usecase"
 	"bootcamp/utils"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
 	"github.com/gorilla/mux"
 )
 
+type pokemonControllerInterface interface {
+	getPokemonCSV(queryParams url.Values) (pokemon model.PokemonList, err error)
+	getPokemonCSVById(id string) (pokemon model.Pokemon, err error)
+	getPokemonThroughApi(name string) (pokemon model.FetchedPokemon, err error)
+	getPokemon() (pokemon model.PokemonList, err error)
+	getPokemonById(id string) (pokemon model.Pokemon, err error)
+	getPokemonFromReader(reader io.ReadCloser) (pokemon model.Pokemon, err error)
+	addPokemon(poke model.Pokemon) (pokemon model.Pokemon, err error)
+	updatePokemon(id string, poke model.Pokemon) (pokemon model.Pokemon, err error)
+	deletePokemon(id string) (pokemon model.Pokemon, err error)
+}
+
+type PokemonController struct{}
+
+func (pc PokemonController) getPokemonCSV(queryParams url.Values) (model.PokemonList, error) {
```

Comment:
This approach for controller seems like duplicating code. Your controller struct should have methods that have the handler signature, so you would be using those methods to handle requests.

Reply:
👌